core/src/main/java/com/netflix/iceberg/hadoop/HadoopTables.java

```diff
@@ -66,6 +70,28 @@ public Table load(String location) {
     return new BaseTable(ops, location);
   }
 
+  @Override
+  public void drop(String location) {
+    final Path path = new Path(location);
+    final FileSystem fs = Util.getFS(path, conf);
+    try {
+      fs.delete(path, true);
+    } catch (IOException e) {
+      throw new RuntimeIOException(e);
+    }
+  }
+
+  @Override
+  public void rename(String location, String newLoaction) {
+    final Path path = new Path(location);
+    final FileSystem fs = Util.getFS(path, conf);
+    try {
+      fs.rename(path, new Path(path, newLoaction));
```

Comment:
The `rename` API returns a boolean. `true` if rename was successful and in `false` in some cases. For instance, if the parent directory is not present in `newLocation` then `rename` will return a `false` instead of throwing an exception. Example: 

```
rename `/a/b/c/x` to `/a/b/d/x` will return `false` if `d` does not exist.

Reply:
It might vary on HSFS implementations (for instance, ADLS) but it would be better to check the return boolean.